spec/ruby/core/numeric/shared/step.rb

```diff
@@ -0,0 +1,397 @@
+require File.expand_path('../../../../spec_helper', __FILE__)
+require File.expand_path('../../helpers/step', __FILE__)
+require File.expand_path('../../fixtures/classes', __FILE__)
+
+describe :numeric_step, :shared => true do
+  before :each do
+    ScratchPad.record []
+    @prc = lambda { |x| ScratchPad << x }
+  end
+
+  it "defaults to step = 1" do
+    1.send(@method, *get_args(@args_type, 5), &@prc)
+    ScratchPad.recorded.should == [1, 2, 3, 4, 5]
+  end
+
+  it "returns an Enumerator when step is 0" do
+    1.send(@method, *get_args(@args_type, 2, 0)).should be_an_instance_of(enumerator_class)
+  end
+
+  it "returns an Enumerator when not passed a block and self > stop" do
+    1.send(@method, *get_args(@args_type, 0, 2)).should be_an_instance_of(enumerator_class)
+  end
+
+  it "returns an Enumerator when not passed a block and self < stop" do
```

Comment:
This and the above spec should probably be wrapped in another `describe` like so as this makes them a bit easier to grok (at least in my opinion):

```ruby
describe 'with a block' do
  it 'returns an Enumerator when step is 0' do

  end

  ...
end
```